src/shared/libfido2-util.c

```diff
@@ -115,3 +117,138 @@ int dlopen_libfido2(void) {
         return 1;
 }
 #endif
+
+int fido2_list_devices(void) {
+#if HAVE_LIBFIDO2
+        _cleanup_(table_unrefp) Table *t = NULL;
+        size_t allocated = 64, found = 0;
+        fido_dev_info_t *di = NULL;
+        int r;
+
+        r = dlopen_libfido2();
+        if (r < 0)
+                return log_error_errno(r, "FIDO2 token support is not installed.");
+
+        di = sym_fido_dev_info_new(allocated);
+        if (!di)
+                return log_oom();
```

Comment:
same question re. log_oom vs -ENOMEM - this is now library-ish code as well, right? (sorry for the piecemeal review but it's a big PR and didn't have time yesterday)

Reply:
well, it's kinda library code, but it's purpose is to dump something on screen in a human friendly way. Given that it's directed to humans anyway it should just log immediately.

i.e. it's more "sharing of parts of main programs" rather than purely library code.

We do this in a similar fashion in other pieces of code that are only used in user-facing stuff already, where it makes little sense to avoid logging given that all users of the code place it in directly user-visible logic anyway.

Reply:
Ok that makes sense, just wanted to make sure as there's a few instances of those with the move from homed to shared